cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller.go

```diff
@@ -610,6 +627,13 @@ func (c *machineController) findScalableResourceProviderIDs(scalableResource *un
 
 		if found {
 			if providerID != "" {
+				// If the machine is deleting, prepend the deletion guard on the provider id
+				// so that it can be properly filtered when counting the number of nodes and instances.
+				if !machine.GetDeletionTimestamp().IsZero() {
+					klog.V(4).Infof("Machine %q has a non-zero deletion timestamp", machine.GetName())
+					providerID = fmt.Sprintf("%s%s_%s", deletingMachinePrefix, machine.GetNamespace(), machine.GetName())
```

Comment:
I'd love to see this entire string concatenation behind a UT-tested func, and then a UT scenario where we invoke the func w/ the 3 inputs, and then test that we always get a true response back from `isDeletingMachineProviderID` when we pass in the return from this new func.

Reply:
i put the function and unit tests there, i just forgot to plug it in. i was having a weird unit test issue.

Reply:
added 2 more test cases, and fixed the commented out line here.